server/internal/decision/engine.go

```diff
@@ -459,6 +459,38 @@ func (e *Engine) WantlistForPeer(p peer.ID) []wl.Entry {
 	return entries
 }
 
+// TotalWants returns the total number of current wants being served
+// across all peers
+func (e *Engine) TotalWants() uint64 {
+	// copying this list to avoid a nested RLock call
+	// maybe unneccesary but seems prudently cautious
+	e.lock.RLock()
+	ledgers := make([]*ledger, 0, len(e.ledgerMap))
+	for _, l := range e.ledgerMap {
+		ledgers = append(ledgers, l)
+	}
+	e.lock.RUnlock()
+
+	total := 0
+	for _, l := range ledgers {
+		l.lk.RLock()
+		total += l.wantList.Len()
+		l.lk.RUnlock()
+	}
+	return uint64(total)
+}
```

Comment:
Allocating a temporary slice, iteration over a map, locking / unlocking a bunch of things.
Maybe this code is very cold ? If it is then this is fine, but it has lots of "I am very slow" redflags. :slightly_smiling_face: 

I feel this could be made much faster if wantlists used `sync/atomic` to increment a shared `*uint64`.